src/nimgen/file.nim

```diff
@@ -54,20 +54,24 @@ proc rename*(file: string, renfile: string) =
     nimout = getNimout(file, false)
     newname = renfile.replace("$nimout", extractFilename(nimout))
 
-  if newname =~ peg"(!\$.)*{'$replace'\s*'('\s*{(!\)\S)+}')'}":
-    var final = nimout.extractFilename()
-    for entry in matches[1].split(","):
-      let spl = entry.split("=")
-      if spl.len() != 2:
-        echo "Bad replace syntax: " & renfile
-        quit(1)
+  if newname.find("replace(") > 0:
+    let matches = findAndCaptureAll(newname, re"\w+=\w+")
```

Comment:
I'm not sure if this is equivalent to the peg in the original. I don't know how pegs work, so @genotrance will have to take a look at this :) Otherwise, PR looks good!

Reply:
I tested on 7z wrapper and gave same results as in pegged version. :-)

Reply:
That might be good enough, but it also might not be. This is why we need some unit tests as well :)